### PR TITLE
Search for tasks based on tags

### LINKED ISSUE #5: View task based on tags
As a lazy university student, I can view the tasks based on category tags, so that I can locate tasks easily.

### PR DESCRIPTION
**Pull request by noelmathewisaac**
Closes #5
- Add command to search for tasks based on tags (1 or more)
- Command format `tag-search [tag1] [tag2 ...]`
- Add tests related to the command

### CODE REVIEW

File: src/test/java/seedu/taskify/model/task/TagContainsKeywordsPredicateTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        TagContainsKeywordsPredicate firstPredicateCopy = new TagContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different task -> returns false
```
Comment (CSmortal) [inline src/test/java/seedu/taskify/model/task/TagContainsKeywordsPredicateTest.java]: change to "different predicate" line 37

File: src/test/java/seedu/taskify/logic/commands/TagSearchCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        TagSearchCommand tagSearchFirstCommandCopy = new TagSearchCommand(firstPredicate);
+        assertTrue(tagSearchFirstCommand.equals(tagSearchFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(tagSearchFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(tagSearchFirstCommand.equals(null));
+
+        // different task -> returns false
```
Comment (CSmortal) [inline src/test/java/seedu/taskify/logic/commands/TagSearchCommandTest.java]: change to "different command" line 53

File: src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    private TagSearchCommandParser parser = new TagSearchCommandParser();
+
+    @Test
+    public void parse_emptyArg_throwsParseException() {
+        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                           TagSearchCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsTagSearchCommand() {
```
Comment (CSmortal) [inline src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java:25]: could add a few more tests just to be safe, for both failure and success, maybe something like "ModuleTutorial" for user input and predicate is {"Module", "Tutorial"}?

**Review by CSmortal — CHANGES_REQUESTED** [review]
could add a few more tests for parsing using TagSearchCommandParser. Have a few additional comments too

File: src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java (hunk 3 above)
Comment (noelmathewisaac) [reply src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java:25]: Ok, will add! The tests I wrote so far follow the same format as the ones for the find command.

**Comment by noelmathewisaac** [comment]
> could add a few more tests for parsing using TagSearchCommandParser. Have a few additional comments too

By additional comments, do you mean comments to explain the tests?

**Comment by CSmortal** [comment]
> 
> 
> > could add a few more tests for parsing using TagSearchCommandParser. Have a few additional comments too
> 
> By additional comments, do you mean comments to explain the tests?

I'm referring to changing the comments for line 37 of TagContainsKeywordsPredicateTest and line 53 of TagSearchCommandTest (just the first two reviews for this PR)

**Comment by noelmathewisaac** [comment]
> > > could add a few more tests for parsing using TagSearchCommandParser. Have a few additional comments too
> > 
> > 
> > By additional comments, do you mean comments to explain the tests?
> 
> I'm referring to changing the comments for line 37 of TagContainsKeywordsPredicateTest and line 53 of TagSearchCommandTest (just the first two reviews for this PR)

ohh okay 👍

File: src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java (hunk 3 above)
Comment (noelmathewisaac) [reply src/test/java/seedu/taskify/logic/parser/TagSearchCommandParserTest.java:25]: The tests in `TagSearchCommandParserTest` checks if the parser splits on space correctly.  The test you mentioned should be part of `TagContainsKeywordsPredicateTest`. I have added it there.

**Review by CSmortal — APPROVED** [review]

**Review by CSmortal — APPROVED** [review]
LGTM

**Review by khiaxeng — APPROVED** [review]
LGTM

**Pull request merged**